rfc/helm-synthesis.md

```diff
@@ -0,0 +1,205 @@
+    * **If library name value is similar**, the generated manifests are the only new changes to be added to the existing chart and there is no changes to the chart structure. This is to accommodate any manual changes made to the chart after its initial generation.
+
+    > Call-Out:
+    The existing manifests in the `templates` folder would be overwritten during synthesis.
+
+
+The synthesis process would resume as normal after the substitution of keys is finished or existing chart.yaml has been validated. And since format was `helm`, the template that is generated by synthesizing is placed in the `<--output value>/templates` folder. And the user can now run,
+
+```
+helm install <release-name> ./<--output value>
+```
+
+to deploy to their Kubernetes cluster.
+
+**User runs `cdk8s synth --format helm`**
+
+This command would execute successfully and will store the chart to the default location. 
+
+For the default directory, since the directory name is suggested to be the chart name by [helm](https://v2.helm.sh/docs/developing_charts/#the-chart-file-structure). So, the `--output` value would be set to the default of library name(if no code synthesis) or directory name(if CDK8s app is local) if `--format` is helm, which is true in this scenario.
+
+### Is this a breaking change?
+
+This is not a breaking change. This is adding new functionality to the CDK8s CLI Synth command. 
+
+### What alternative solutions did you consider?
+
+* **`helm create` command to generate chart structure**
+
+  [helm create](https://helm.sh/docs/helm/helm_create/) is a helm command that generates a templated chart with the necessary folder structure and files needed to deploy the chart with helm. I did not use this to create the chart structure for two reasons,
+  1. This would mean that we need to take a dependency with helm or mention to the user that helm needs to be present on their machine for the synthesis to work in this scenario.
+  2. This command creates all the necessary structure but also creates files for simple `nginx` application. For our purposes, these files would either needed to be deleted or modified to work with the manifests we are generating. This feels like an over kill to accomplish our requirements.
+
+* **[helm-x](https://github.com/mumoshu/helm-x) and [chartify](https://github.com/helmfile/chartify) libraries**
+  
+  One of the functionality that these libraries provide users is to convert Kubernetes resource YAMLs to Helm Charts. These can be used to generate helm charts from our generated manifests from CDK8s app. Currently, these are version `0.x.x`, which means adding these to support our feature could result in added maintenance load for the team. I believe adding a solution of our own would give us more control and not limit us to a library's functionality.
```

Comment:
I suppose another reason to reject is because it adds a dependency on a binary file, similar to `helm create`